File: typescript/examples/info_vaults.ts
#!/usr/bin/env npx ts-node
// @ts-nocheck
/**
 * Vaults & Delegation Example
 *
 * Shows how to query vault information and user delegations.
 *
 * Usage:
 *     export QUICKNODE_ENDPOINT="https://your-endpoint.hype-mainnet.quiknode.pro/YOUR_TOKEN"
 *     export USER_ADDRESS="0x..."  # Optional
 *     npx ts-node info_vaults.ts
 */

import { HyperliquidSDK } from '@quicknode/hyperliquid-sdk';

const ENDPOINT = process.env.QUICKNODE_ENDPOINT;
const USER = process.env.USER_ADDRESS;

if (!ENDPOINT) {
  console.error("Set QUICKNODE_ENDPOINT environment variable");
  process.exit(1);
}

async function main() {
  // Single SDK instance — access everything through sdk.info, sdk.core, sdk.evm, etc.
  const sdk = new HyperliquidSDK(ENDPOINT);
  const info = sdk.info;

  console.log("=".repeat(50));
  console.log("Vaults & Delegation");
  console.log("=".repeat(50));

  // Vault summaries
  console.log("\n1. Vault Summaries:");
  const vaults = await info.vaultSummaries();
  console.log(`   Total: ${vaults.length}`);
  for (const v of vaults.slice(0, 3)) {
    console.log(`   - ${v.name || 'N/A'}: TVL $${v.tvl || '?'}`);
  }

  // User delegations
  if (USER) {
    console.log(`\n2. Delegations (${USER.slice(0, 10)}...):`);
    const delegations = await info.delegations(USER);
    if (delegations.length > 0) {
      console.log(`   ${delegations.length} active`);
    } else {
      console.log("   None");
    }
  } else {
    console.log("\n(Set USER_ADDRESS for delegation info)");
  }

  console.log("\n" + "=".repeat(50));
  console.log("Done!");
}

main().catch(console.error);

File: python/examples/info_vaults.py
#!/usr/bin/env python3
"""
Vaults & Delegation Example

Shows how to query vault information and user delegations.

Requirements:
    pip install hyperliquid-sdk

Usage:
    export ENDPOINT="https://your-endpoint.hype-mainnet.quiknode.pro/TOKEN"
    export USER_ADDRESS="0x..."  # Optional
    python info_vaults.py
"""

import os
import sys

from hyperliquid_sdk import HyperliquidSDK

ENDPOINT = os.environ.get("ENDPOINT") or os.environ.get("QUICKNODE_ENDPOINT")
USER = os.environ.get("USER_ADDRESS")

if not ENDPOINT:
    print("Error: Set ENDPOINT environment variable")
    print("  export ENDPOINT='https://your-endpoint.hype-mainnet.quiknode.pro/TOKEN'")
    sys.exit(1)


def main():
    print("=" * 50)
    print("Vaults & Delegation")
    print("=" * 50)

    # Single SDK instance - access everything through sdk.info(), sdk.core(), sdk.evm()
    sdk = HyperliquidSDK(ENDPOINT)
    info = sdk.info()

    # Vault summaries
    print("\n1. Vault Summaries:")
    vaults = info.vault_summaries()
    print(f"   Total: {len(vaults)}")
    for v in vaults[:3]:
        name = v.get("name", "N/A")
        tvl = v.get("tvl", "?")
        print(f"   - {name}: TVL ${tvl}")

    # User delegations
    if USER:
        print(f"\n2. Delegations ({USER[:10]}...):")
        delegations = info.delegations(USER)
        if delegations:
            print(f"   {len(delegations)} active")
        else:
            print("   None")
    else:
        print("\n(Set USER_ADDRESS for delegation info)")

    print()
    print("=" * 50)
    print("Done!")


if __name__ == "__main__":
    main()
